### ISSUE TITLE
After deactivating "Flight Mode" the SIM card does not reconnect to the telecom provider  

### ISSUE THREAD

**Issue by Stevie00911** [HelloVolla/ubuntu-touch-beta-tests#71]
Ubuntu 16.04 (2020-W44) release candidate:
After activating and deactivating "Flight Mode" the SIM card does not reconnect to the telecom provider.
The SIM 1 status shows: "searching" for a short while and than stay on "no radio reception".
A rebbot is necessary to get back the reconnection to the telecom provider.

**Comment by JamiKettunen** [comment]
I've seen this issue earlier here as well and I was never able to reproduce it myself; could you run these to enable verbose Ofono logging and run `grep -i 'ofono' /var/log/syslog` later to get logs when you encounter this issue again (you can try toggling Flight Mode a few times just to be sure the issue logged)? That would greatly help with debugging this issue:
```
mount | grep -q ' / .*ro' && sudo mount -o remount,rw /
sudo sed 's/ofonod/ofonod -d/' -i /etc/init/ofono.override
sudo initctl restart ofono
```
To disable verbose logs again you can run the same commands but use the following `sed`: `sudo sed 's/ofonod -d/ofonod/' -i /etc/init/ofono.override`

**Comment by Stevie00911** [comment]
Oh, this sounds complicated.
I gave it a try by using the Terminal application. 

I first typed in:
mount | grep -q ' / .*ro' && sudo mount -o remount,rw /
sudo sed 's/ofonod/ofonod -d/' -i /etc/init/ofono.override
sudo initctl restart ofono

Than I swichted Flight Mode on/off 3 times

Than I typed in the terminal:
grep -i 'ofono' /var/log/syslog

and got these last results, I think this is one Flight mode on/off sequenz:

Nov 23 11:07:23 ubuntu-phablet ofonod[18419]: data reg changed 1 -> 0 (unregistered) attached 0
Nov 23 11:07:47 ubuntu-phablet ofonod[18419]: nw selection is already auto
Nov 23 11:07:47 ubuntu-phablet ofonod[18419]: data reg changed 0 -> 2 (searching) attached 0
Nov 23 11:07:47 ubuntu-phablet ofonod[18419]: data reg changed 2 -> 0 (unregistered) attached 0
Nov 23 11:07:47 ubuntu-phablet ofonod[18419]: data reg changed 0 -> 2 (searching) attached 0
Nov 23 11:07:49 ubuntu-phablet ofonod[18419]: data reg changed 2 -> 4 (unknown) attached 0
Nov 23 11:07:50 ubuntu-phablet ofonod[18419]: data reg changed 4 -> 1 (registered) attached 0

Is this the log you are looking for?

**Comment by gbdomubpkm** [comment]
With the fix that seems to have done miracles (bouygues SIM) for me at the moment, 

/usr/share/ofono/scripts/set-context-property 0 Protocol IP

i can't reproduce this problem (I'm on RC 7 currently).

**Comment by JamiKettunen** [comment]
@Stevie00911 can you attempt reproducing this under the latest RC channel build? This should hopefully be fixed in OTA-16 thanks to [NotKit's binder patch](https://github.com/HelloVolla/android_kernel_volla_mt6763/commit/ade5d98) :)

**Comment by Stevie00911** [comment]
@JamiKettunen,
thank you for your message.
Under Ubuntu 16.04 (2021-W10) release candidate (should be the latest RC channel build) I still have the same behaviour.
So let's hope for OTA-16 :-)

**Comment by Stevie00911** [comment]
Ubuntu 16.04 (OTA-16)
The issue is still present.

**Comment by wurzer** [comment]
I just checked it with OTA 16 Stable. 

1. Enabling flight mode
2. Waiting a couple of minutes
3. Disabling flight mode
4. You can see, that the device is searching for the network of each SIM card
5. Network is connected

It works fine in Germany with Blau.de and Klarmobil

**Comment by Stevie00911** [comment]
Ubuntu 16.04 (OTA-16):
I cannot confirm with my Vollaphone and a SIM card from Germany's Telekom.de inside: 

    1. Enabling flight mode
    2. Waiting a couple of minutes
    3. Disabling flight mode
    4. You can see, that the device is searching for the network of each SIM card
    5. "Kein Empfang", or in English: "No reception" and an empty trinagle with a small cross is displayed
    6.  Reboot: Network is connected

**Comment by wurzer** [comment]
Are you sure, that you are on the stable channel?

**Comment by Stevie00911** [comment]
Yes, I am on the stable channel OTA-16.

**Comment by wurzer** [comment]
Then there must be something different to my device. Are you abroad using roaming?

**Comment by Stevie00911** [comment]
I am using the Vollaphone here from Munich, Germany without roaming.

**Comment by wurzer** [comment]
Could you re-check the reported behavior. It should be fixed at least in the development OTA channel.

**Comment by Stevie00911** [comment]
@wurzer, thank you for your message.
I changed now to the development OTA channel and istalled the "2021-05-14" version.
Here I realized different behaviours depending on the activated Connection type "2G/3G" or "2G/3G/4G".

**Connection type "2G/3G":**
After "Flight Mode" off still no reconnect to the telecom provider, see screen.

![grafik](https://user-images.githubusercontent.com/45417818/118356225-76450280-b574-11eb-84e3-2574258d61ff.png)

**Connection type "2G/3G/4G":**
After "Flight Mode" off the phone reconnects to the Telecom provider, super!
I tested this many times and it seems to work permanently. 
I did not test this Connection type before, because the cellular data function still does not connect to the Internet.
For using the internet, I must switch to the Connection type "2G/3G" again.

So my workaround is actually: 
- Working under Connection type "2G/3G"
- At night turn "Fligt Mode" on
- In the morning turn "Fligt Mode" off
- Swith to Connection type "2G/3G/4G" and get the Telekom connection
- Swith to Connection type "2G/3G" to get the internet connection

What is now needed is to get the internet connection working under Connection type "2G/3G/4G".
This gets more and more imprtand because this year the German Telekom will turn off the 3G net.
